cmd/root.go

```diff
@@ -0,0 +1,134 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra/doc"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/k0sproject/k0s/pkg/build"
+	"github.com/k0sproject/k0s/pkg/util"
+)
+
+var cfgFile string
+var debug bool
+var cmdLogLevels map[string]string
+var logging map[string]string
+
+var defaultLogLevels = map[string]string{
+	"etcd":                    "info",
+	"containerd":              "info",
+	"konnectivity-server":     "1",
+	"kube-apiserver":          "1",
+	"kube-controller-manager": "1",
+	"kube-scheduler":          "1",
+	"kubelet":                 "1",
+}
+
+func init() {
+	rootCmd.PersistentFlags().StringVarP(&cfgFile, "config", "c", "", "config file (default is $HOME/k0s.yaml)")
+	rootCmd.PersistentFlags().BoolVarP(&debug, "debug", "d", false, "Debug logging (default is false)")
+	rootCmd.PersistentFlags().StringToStringVarP(&cmdLogLevels, "logging", "l", defaultLogLevels, "Logging Levels for the different components")
+
+	logging = setLogging(cmdLogLevels)
+
+	// initialize configuration
+	err := initConfig()
+	if err != nil {
+		fmt.Printf("err: %v", err)
+	}
+
+	rootCmd.AddCommand(versionCmd)
+	rootCmd.AddCommand(configCmd)
+	rootCmd.AddCommand(tokenCmd)
+	rootCmd.AddCommand(serverCmd)
+	rootCmd.AddCommand(workerCmd)
+	rootCmd.AddCommand(APICmd)
+	rootCmd.AddCommand(etcdCmd)
+	rootCmd.AddCommand(docs)
+}
+
+var (
+	rootCmd = &cobra.Command{
+		Use:   "k0s",
+		Short: "k0s - Zero Friction Kubernetes",
+		Long: `k0s is yet another Kubernetes distro. Yes. But we do some of the things pretty different from other distros out there.
+It is a single binary all-inclusive Kubernetes distribution with all the required bells and whistles preconfigured to make 
+building a Kubernetes clusters a matter of just copying an executable to every host and running it.`,
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			// set DEBUG from env, or from command flag
+			if viper.GetString("debug") != "" || debug {
+				logrus.SetLevel(logrus.DebugLevel)
+			}
+		},
+	}
+
+	versionCmd = &cobra.Command{
+		Use:   "version",
+		Short: "Print the k0s version",
+
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println(build.Version)
+		},
+	}
+
+	docs = &cobra.Command{
+		Use:   "docs",
+		Short: "Generate Markdown docs for the k0s binary",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := generateDocs()
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+)
+
+func initConfig() error {
+	if cfgFile == "" {
+		home, err := util.HomeDir()
+		if err != nil {
+			return err
+		}
+		cfgFile = filepath.Join(home, "k0s.yaml")
```

Comment:
This actually changes the behaviour to try to load the config file from users home by default. IMO not good idea, not very common place to look configs from. I'd rather preserve the existing behaviour where we try to load it from the current dir.

Reply:
fixed